processing/src/main/java/io/druid/query/timeseries/TimeseriesQuery.java

```diff
@@ -211,17 +226,19 @@ public boolean equals(Object o)
     if (granularity != null ? !granularity.equals(that.granularity) : that.granularity != null) {
       return false;
     }
-    if (postAggregatorSpecs != null ? !postAggregatorSpecs.equals(that.postAggregatorSpecs) : that.postAggregatorSpecs != null) {
+    if (aggregatorSpecs != null ? !aggregatorSpecs.equals(that.aggregatorSpecs) : that.aggregatorSpecs != null) {
       return false;
     }
-
-    return true;
+    return postAggregatorSpecs != null
```

Comment:
IntelliJ can automatically generate `Objects.equals()` (in code generation wizard menu, there is such option in dropdown menu)